### PR TITLE
Use gpu-buffer for logits

### PR DESCRIPTION
**Pull request by qjia7**
With this change, the avg time becomes ~13ms from ~26ms on my ADL with io_binding=1

### CODE REVIEW

**Comment by qjia7** [comment]
@guschmue Please take a look, thanks.

**Comment by qjia7** [comment]
Close it since I used the wrong time after specifying all outputs on gpu. The accurate time should be after post_process if I move logits to gpu.
